catalog/transformer/catalog-transformer-common/src/main/java/ddf/catalog/transformer/common/tika/TikaMetadataExtractor.java

```diff
@@ -13,55 +13,94 @@
 import org.apache.tika.parser.ParseContext;
 import org.apache.tika.parser.Parser;
-import org.xml.sax.ContentHandler;
+import org.apache.tika.sax.BodyContentHandler;
+import org.apache.tika.sax.TeeContentHandler;
+import org.apache.tika.sax.WriteOutContentHandler;
 import org.xml.sax.SAXException;
 
-import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transformer.common.tika.handler.XmlMetadataContentHandler;
 
 public class TikaMetadataExtractor {
-    private final Parser parser;
 
-    private final ContentHandler handler;
+    private final BodyContentHandler bodyHandler;
+
+    private final XmlMetadataContentHandler metadataHandler;
+
+    private final WriteOutContentHandler writeOutContentHandler;
+
+    private final InputStream inputStream;
+
+    private Metadata metadata;
 
     /**
-     * Creates a new {@code TikaMetadataExtractor} with a {@link Parser} and {@link ContentHandler}.
+     * Constructs a new tika extractor which parses the provided input stream
+     * into a tika Metadata object, the body text, and the metadata XML
      *
-     * @param parser  the {@code Parser} to use
-     * @param handler the {@code ContentHandler} to use with {@code parser}
+     * @param inputStream - the input stream to be parsed
+     * @throws TikaException - if parsing fails
      */
-    public TikaMetadataExtractor(final Parser parser, final ContentHandler handler) {
-        this.parser = parser;
-        this.handler = handler;
+    public TikaMetadataExtractor(InputStream inputStream) throws TikaException {
+        this(inputStream, -1);
     }
 
     /**
-     * Parses metadata from {@code inputStream} using the supplied {@link Parser},
-     * {@link ContentHandler}, and {@link ParseContext}.
+     * Constructs a new tika extractor which parses the provided input stream
+     * into a tika Metadata object, the body text, and the metadata XML
+     * The body text is truncated after maxLength
      *
-     * @param inputStream  the data to parse
-     * @param parseContext context information to pass to the {@code Parser}, may be null
-     * @return a {@link Metadata} object containing the metadata that the {@code Parser} was able to
-     * extract from {@code inputStream}
-     * @throws CatalogTransformerException
-     * @throws IOException
+     * @param inputStream - the input stream to be parsed
+     * @param maxLength   - the max length of the parsed body text
+     * @throws TikaException - if parsing fails
      */
-    public Metadata parseMetadata(final InputStream inputStream, final ParseContext parseContext)
-            throws CatalogTransformerException, IOException {
-        final Metadata metadata = new Metadata();
+    public TikaMetadataExtractor(InputStream inputStream, int maxLength) throws TikaException {
+        notNull(inputStream);
+        this.inputStream = inputStream;
+        this.metadataHandler = new XmlMetadataContentHandler(StandardCharsets.UTF_8.toString());
```

Comment:
This can cause the system to blow up with an out of memory error if you are not providing a stream that is backed by a file.

Reply:
Changed to use a TemporaryFileBackedOutputStream with XmlMetadataContentHandler